### PR TITLE
test: replace as assertions

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
1. Replace `as` assertions with `partial` for `BranchUpgradeConfig` 

## Context
 - Ref : https://github.com/renovatebot/renovate/issues/19856

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/updates/generate.spec.ts [hunk 1]
```diff
@@ -1166,7 +1171,7 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('dedupes duplicate table rows', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:1174]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 2]
```diff
@@ -1133,19 +1137,20 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('prevents issue with duplicating "v" character', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:1140]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 3]
```diff
@@ -1077,7 +1081,7 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('merge excludeCommitPaths if appears in upgrade', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:1084]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 4]
```diff
@@ -1036,14 +1040,14 @@ describe('workers/repository/updates/generate', () => {
...
-      ];
+      ]);
       const res = generateBranchConfig(branch);
       expect(res.pendingChecks).toBeUndefined();
       expect(res.upgrades).toHaveLength(1);
     });
 
     it('displays pending versions', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:1050]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 5]
```diff
@@ -1013,14 +1017,14 @@ describe('workers/repository/updates/generate', () => {
...
-      ];
+      ]);
       const res = generateBranchConfig(branch);
       expect(res.pendingChecks).toBeTrue();
       expect(res.upgrades).toHaveLength(2);
     });
 
     it('filters pendingChecks', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:1027]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 6]
```diff
@@ -988,15 +992,15 @@ describe('workers/repository/updates/generate', () => {
...
+      ]);
       const res = generateBranchConfig(branch);
       expect(
         res.upgrades.map((upgrade) => upgrade.fileReplacePosition)
       ).toStrictEqual([undefined, undefined, 4, 1]);
     });
 
     it('passes through pendingChecks', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:1003]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 7]
```diff
@@ -949,13 +953,13 @@ describe('workers/repository/updates/generate', () => {
...
         },
-      ];
+      ]);
       const res = generateBranchConfig(branch);
       expect(res.prBodyColumns).toEqual(['column-a', 'column-b', 'column-c']);
     });
 
     it('sorts upgrades, without position first', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:962]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 8]
```diff
@@ -938,7 +942,7 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('combines prBodyColumns', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:945]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 9]
```diff
@@ -823,7 +827,7 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('handles @types specially (reversed)', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:830]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 10]
```diff
@@ -758,7 +762,7 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('handles @types specially', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:765]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 11]
```diff
@@ -466,7 +466,7 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('fixes different messages', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:469]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```
do we really need the partial here? isn't satisfies better here?

File: lib/workers/repository/updates/generate.spec.ts [hunk 12]
```diff
@@ -409,7 +409,7 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('groups multiple digest updates', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:412]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 13]
```diff
@@ -366,7 +366,7 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('groups multiple upgrades different value but same version', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:369]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 14]
```diff
@@ -323,7 +323,7 @@ describe('workers/repository/updates/generate', () => {
     });
 
     it('groups multiple upgrades different version but same value', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:326]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/updates/generate.spec.ts [hunk 15]
```diff
@@ -263,14 +263,14 @@ describe('workers/repository/updates/generate', () => {
...
-      ];
+      ]);
       const res = generateBranchConfig(branch);
       expect(res.groupName).toBeDefined();
       expect(res.recreateClosed).toBeTrue();
     });
 
     it('groups multiple upgrades different version', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:273]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```

File: lib/workers/repository/update/pr/changelog/release-notes.spec.ts [hunk 16]
```diff
@@ -97,11 +97,12 @@ describe('workers/repository/update/pr/changelog/release-notes', () => {
     it('returns input if invalid', async () => {
       const input = { a: 1 };
       expect(
-        await addReleaseNotes(input as never, {} as BranchUpgradeConfig)
+        await addReleaseNotes(input as never, partial<BranchUpgradeConfig>({}))
```
Comment (viceice) [inline lib/workers/repository/update/pr/changelog/release-notes.spec.ts:100]: we should allow partial function call without empty object, like `partial<Type>()`.

File: lib/workers/repository/updates/generate.spec.ts [hunk 17]
```diff
@@ -15,7 +15,7 @@ beforeEach(() => {
 describe('workers/repository/updates/generate', () => {
   describe('generateBranchConfig()', () => {
     it('does not group single upgrade', () => {
-      const branch: BranchUpgradeConfig[] = [
+      const branch: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/updates/generate.spec.ts:18]:
```suggestion
      const branch = partial<BranchUpgradeConfig>([
```


please do same for others below

Comment (RahulGautamSingh) [reply lib/workers/repository/updates/generate.spec.ts:18]: Could you comment on the [TBD] change suggestion in the PR description

**Comment by viceice** [comment]
sounds good, go ahead with 2. but make small PR's for easier and faster review.

**Comment by RahulGautamSingh** [comment]
Blocked by https://github.com/renovatebot/renovate/pull/20618

**Comment by RahulGautamSingh** [comment]
> 2\. [TBD] Replace `defaultConfig = getConfig` -> smaller configs which are narrowed down to contain only the options needed for the specific test

Tracking it here https://github.com/renovatebot/renovate/issues/20619

**Comment by RahulGautamSingh** [comment]
> sounds good, go ahead with 2. but make small PR's for easier and faster review.
Tracking it here https://github.com/renovatebot/renovate/issues/20619

File: lib/workers/repository/update/pr/changelog/release-notes.spec.ts (hunk 16 above)
Comment (RahulGautamSingh) [reply lib/workers/repository/update/pr/changelog/release-notes.spec.ts:100]: https://github.com/renovatebot/renovate/pull/20618

File: lib/workers/repository/update/branch/execute-post-upgrade-commands.spec.ts [hunk 18]
```diff
@@ -10,7 +10,7 @@ jest.mock('../../../../util/git');
 describe('workers/repository/update/branch/execute-post-upgrade-commands', () => {
   describe('postUpgradeCommandsExecutor', () => {
     it('handles an artifact which is a directory', async () => {
-      const commands: BranchUpgradeConfig[] = [
+      const commands: BranchUpgradeConfig[] = partial<BranchUpgradeConfig>([
```
Comment (viceice) [inline lib/workers/repository/update/branch/execute-post-upgrade-commands.spec.ts:13]:
```suggestion
      const commands = partial<BranchUpgradeConfig>([
```
please fix them all

**Review by viceice — CHANGES_REQUESTED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**